.github/workflows/tests.yml

```diff
@@ -45,11 +45,14 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           cache: pip
-          cache-dependency-path: setup.cfg
+          cache-dependency-path: |
+            setup.cfg
+            pyproject.toml
```

Comment:
I though the problem was coming from a caching problem, but in the end, adding an explicit install of `importlib_metadata` was what fixed the problem.